### PR TITLE
Add resource and handlers for pseudo- legacy mlab-ns support

### PR DESCRIPTION
**Pull request by stephen-soltesz**

### CODE REVIEW

**Comment by stephen-soltesz** [comment]
Example invocations:

```
$ curl --dump-header - 'https://locate-dot-mlab-sandbox.appspot.com/ndt?policy=geo_options'
[{"city":"New York","country":"US","fqdn":"ndt-mlab4-lga0t.mlab-sandbox.measurement-lab.org","site":"lga0t"},{"city":"New York","country":"US","fqdn":"ndt-mlab1-lga1t.mlab-sandbox.measurement-lab.org","site":"lga1t"},{"city":"San Francisco Bay Area","country":"US","fqdn":"ndt-mlab1-nuq0t.mlab-sandbox.measurement-lab.org","site":"nuq0t"}]

$ curl --dump-header - 'https://locate-dot-mlab-sandbox.appspot.com/ndt'
{"city":"New York","country":"US","fqdn":"ndt-mlab2-lga0t.mlab-sandbox.measurement-lab.org","site":"lga0t"}

$ curl --dump-header - 'https://locate-dot-mlab-sandbox.appspot.com/ndt_ssl'
{"city":"New York","country":"US","fqdn":"ndt-mlab1-lga0t.mlab-sandbox.measurement-lab.org","site":"lga0t"}

$ curl --dump-header - 'https://locate-dot-mlab-sandbox.appspot.com/ndt_ssl?format=bt'
New York, US|ndt-mlab2-lga0t.mlab-sandbox.measurement-lab.org

$ curl --dump-header - 'https://locate-dot-mlab-sandbox.appspot.com/ndt_ssl?format=bt&policy=geo_options'
New York, US|ndt-mlab4-lga0t.mlab-sandbox.measurement-lab.org
San Francisco Bay Area, US|ndt-mlab1-nuq0t.mlab-sandbox.measurement-lab.org
```

**Review by cristinaleonr — COMMENTED** [review]
The changes look fine, but there is a non-trivial amount of complexity/translation in `handler_legacy.go`. Why are we adding mlab-ns resources to the Locate? Do we know what percentage of users target mlab-ns? I thought we were going to stop supporting it ([blog](https://www.measurementlab.net/blog/retiring-ndt5-raw/)). 

> *[Reviewable](https://reviewable.io/reviews/m-lab/locate/184#-:-Nr1t9M8AITolk-DxM-A:b-h1bpjf)* status: 0 of 1 approvals obtained

**Review by stephen-soltesz — COMMENTED** [review]
Your question is good. I'm trying to do the right thing. The following are all true:
* App Engine has ended support and deployment for python2.7 apps, though they can continue running (like mlab-ns)
* mlab-ns still depends on old naming conventions for ndt servers, i.e. ndt-iupui, stalling changes to siteinfo & prometheus-support
* ~90% of requests for ndt5/raw go to mlab-ns (/ndt or /ndt_ssl) vs Locate v2 (/v2/nearest/ndt/ndt5)
* It looks like ndt5+raw is the second most common protocol in the data after ndt7+wss (admittedly a distant second, 91% vs ~3.5%)
* Our own tooling (ndt5-client-go) still use mlab-ns
* While ~50% of mlab-ns requests are from identifiable client integrations, M-Lab has not had time yet to reach out to any about migration.

So, rather than rip the bandaid off, I'm trying to provide an intermediate path that allows us to disable the old App Engine code base in exchange for a much smaller handler which we can control how we want (e.g. to reduce availability gradually to encourage migration). Also, the total code under maintenance reduces ~20x from ~10141 loc vs ~500 loc with tests, so this would be a net reduction.

In summary: net code reduction, more control for us operationally, preserve some backward compatibility for existing users until M-Lab can coordinate the migration and registration process.

This was the original evaluation, the queries dates are a little dated now, but the same trends seem to be there today: https://docs.google.com/document/d/1Tr5ImBK86_GWNFLlzuVvbG6EegodLAiCXMxQlmBU6vA/edit

> *[Reviewable](https://reviewable.io/reviews/m-lab/locate/184#-:-Nr6aJcdDUuX8dF579Gb:bn73c2i)* status: 0 of 1 approvals obtained

**Comment by stephen-soltesz** [comment]
PTAL?

**Review by cristinaleonr — APPROVED** [review]
:lgtm: 

It might be worth creating a github issue that you can reference in the removal TODOs.

> *[Reviewable](https://reviewable.io/reviews/m-lab/locate/184#-:-NrGeXVk8cmOL_U3a9sB:bg6x1b2)* status: <span>:shipit:</span> complete! 1 of 1 approvals obtained

File: handler/handler_legacy.go
Comment (robertodauria) [inline handler/handler_legacy.go]: Since this PR was written, requirements have significantly changed: in particular, we now have only one integrator that needs legacy support, and they are using mlab-ns in a specific way - we don't need to support the full option set, nor I think we should.

File: handler/handler_legacy.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	case "geo_options":
+		// all results
+		break
+	default:
+		results = results[:1]
+	}
+
+	// Default format is JSON.
+	switch q.Get("format") {
+	case "bt":
```
Comment (robertodauria) [inline handler/handler_legacy.go]: We won't support any client still using this, right? I think it could be removed.
